src/java/org/apache/cassandra/index/sai/utils/RangeUnionIterator.java

```diff
@@ -46,6 +47,11 @@ public PrimaryKey computeNext()
         RangeIterator candidate = null;
         for (RangeIterator range : ranges)
         {
+            if (lastKey != null)
```

Comment:
I think the changes to RUI aren't necessary if the sub-iterators are fulfilling their own deduplcation contract

Reply:
You're right that this isn't be necessary. I added it while I was debugging and it appeared useful later on because on line 73, I replaced `next()` with `next`, which doesn't advance the iterator. Adding the missed `()` shows that `range.peek().compareTo(lastKey) == 0` is never true.